src/main/java/us/abstracta/wiresham/VirtualTcpClient.java

```diff
@@ -59,26 +40,38 @@ private Socket buildSocket()
 
   public void run() {
     try {
-      new ConnectionFlowDriver(buildSocket(), readBufferSize, flow).run();
-    } catch (NoSuchAlgorithmException | KeyManagementException | IOException e) {
+      for (List<PacketStep> packetSteps : packetsByPortMap.keySet()) {
```

Comment:
maybe is not a map, and just a list of entities with socket and list of packets? Like Socket packets, or something like that? But sounds like that sequentiality would not be respected, since "clients" would be executed in concurrency, wouldn't they?

Reply:
Oh, I see you use a singleThreadExecutor... maybe that discards the potential problem with sequentiality

Reply:
But in fact, it doesn't, because your are not assured of the order of tasks submitted to a execution service :(. They will run, but not necessarily in the same order they were submitted (as far as I remember)